### PR TITLE
feat: Quiz System and Data Storing with Google Spreadsheet

### PR DESCRIPTION
**Pull request by fahlisaputra**
Issue #5 

**Added:**
- Quiz route `/quiz/**`
- `quizzes` directory contains questions
- Page for quiz system, copying Kidspiration
- Metadata for each question file (e.g. title, author, description)
- Index page quiz
- Report API to Spreadsheets

**Changes:**
- Install uuid package
- Install googleapis package
- Move quiz.jsx to `/quiz` directory as `index.jsx`

**Doneness:**
- [x] Build is okay - I made sure that this change is building successfully.
- [x] No Bugs - I made sure that this change is working properly as expected. It doesn't have any bugs that you are aware of.

**Not Implementeđ:**
Showing image on question if needed.

### CODE REVIEW

File: quizzes/firebase.json [hunk 1]
```diff
@@ -0,0 +1,48 @@
+{
+    "metadata": {
```
Comment (vzrenggamani) [inline quizzes/firebase.json:2]: I think we should try to add global identifier for Quiz and Question itself like
```json
{
"metadata": {
        "uuid": "fb833dbe-c98b-45b1-b054-749a2ae77a05", //global identifier
        "title": "Firebase",
        "author": "Fahli Saputra",
        "description": "short description",
        "thumbnail": "https://fahli.net/wp-content/uploads/2020/07/Instalasi-Debian-9-dan-Konfigurasi-IP.jpg"
    },
}
```

```json
{
            "uuid": "fb833dbe-c98b-45b1-b054-749a2ae77a05" //global identifier
            "question": "What is the capital of India?",
            "image": "",
            "options": [
                {
                    "answer": "Delhi",
                    "isCorrect": true
                },
                {
                    "answer": "Mumbai"
                },
                {
                    "answer": "Kolkata"
                },
                {
                    "answer": "Chennai"
                }
            ]
        },
```

File: src/pages/quiz/[slug].jsx [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+    }
+}
+
+const Quiz = (questions) => {
+    const [currentQuestion, setCurrentQuestion] = React.useState(0);
+    const [score, setScore] = React.useState(0);
+    const [showScore, setShowScore] = React.useState(false);
+
+    // function to handle every question 
+    const handleQuestion = (answer) => {
```
Comment (vzrenggamani) [inline src/pages/quiz/[slug].jsx:40]: I wonder could we add handler for this function to send a POST request when the users choose wrong question?

```
if (user.answer IS CORRECT) {
givethemscore(1)
} else {
POST https://api.db?quizid=UUID&questionID=UUID&timestamp=TIMESTAMP
}
```

**Review by vzrenggamani — CHANGES_REQUESTED** [review]
RFC:
- Add quiz identifier
- Add Question Identifier
- Add Handler to throw user wrong to an mock api


Anyway, How can we share the API Key for development?

File: quizzes/firebase.json (hunk 1 above)
Comment (fahlisaputra) [reply quizzes/firebase.json:2]: done on commit `d11af97cb72fcbfdda8b7fa480ab181d77000186`

**Review by vzrenggamani — APPROVED** [review]
Worked on staging, should be OK for pushing to the main branch.

**Review by vzrenggamani — CHANGES_REQUESTED** [review]
- [x] Please add .env.example
- [ ] Please move the `quiz.jsx` to the quiz folder
- [ ] Please rename `store-report` API to `noun` like `report`. Store report is already known as POST /report

**Comment by fahlisaputra** [comment]
> * [x]  Please add .env.example
> * [x]  Please move the `quiz.jsx` to the quiz folder
> * [x]  Please rename `store-report` API to `noun` like `report`. Store report is already known as POST /report

done, commit `6c97bb8343cff312e75aae3e9ee7564e86f38f08`

**Review by vzrenggamani — APPROVED** [review]
Looks Good!

**Pull request merged**